File: src/main/java/com/kasisoft/libs/common/annotation/Prio.java
package com.kasisoft.libs.common.annotation;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.*;

import java.lang.annotation.*;

/**
 * This annotation allows to provide some ordering to class types.
 * 
 * @author daniel.kasmeroglu@kasisoft.net
 */
@Retention(RUNTIME)
@Target({TYPE})
public @interface Prio {

    int value() default 0;

} /* ENDINTERFACE */

File: src/main/kotlin/com/kasisoft/libs/common/annotation/Prio.kt
package com.kasisoft.libs.common.annotation

/**
 * This annotation allows to provide some ordering to class types.
 *
 * @author daniel.kasmeroglu@kasisoft.com
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class Prio(val value: Int = 0)
